tst/CTA.WebForms2Blazor.Tests/ControlConverters/TextBoxControlConverterTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿using CTA.WebForms2Blazor.ControlConverters;
+using HtmlAgilityPack;
+using NUnit.Framework;
+
+namespace CTA.WebForms2Blazor.Tests.ControlConverters
+{
+    public class TextBoxControlConverterTests
+    {
+        private TextBoxControlConverter _testConverter;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _testConverter = new TextBoxControlConverter();
+        }
+
+        [Test]
+        public void Convert2Blazor_Returns_Standard_Input_For_SingleLine_Text_Mode()
+        {
+            var htmlString = @"<asp:TextBox ID=""TestId"" Text=""Example Text"">";
+            var expectedString = @"<input id=""TestId"" value=""Example Text"">";
+
+            var htmlNode = HtmlNode.CreateNode(htmlString, ControlConverter.PreserveCapitalization);
+            var convertedNode = _testConverter.Convert2Blazor(htmlNode);
+            var actualString = convertedNode.WriteTo();
+
+            Assert.AreEqual(expectedString, actualString);
+        }
+
+        [Test]
+        public void Convert2Blazor_Returns_TextArea_Input_For_MultiLine_Text_Mode()
+        {
+            var htmlString = @"<asp:TextBox ID=""TestId"" TextMode=""MultiLine"" Text=""Example Text"">";
+            var expectedString = @"<textarea id=""TestId"">Example Text</textarea>";
+
+            var htmlNode = HtmlNode.CreateNode(htmlString, ControlConverter.PreserveCapitalization);
+            var convertedNode = _testConverter.Convert2Blazor(htmlNode);
+            var actualString = convertedNode.WriteTo();
+
+            Assert.AreEqual(expectedString, actualString);
+        }
+
+        [Test]
+        public void Convert2Blazor_Sets_Password_Type_For_Password_Text_Mode()
+        {
+            var htmlString = @"<asp:TextBox ID=""TestId"" TextMode=""Password"">";
+            var expectedString = @"<input id=""TestId"" type=""password"">";
+
+            var htmlNode = HtmlNode.CreateNode(htmlString, ControlConverter.PreserveCapitalization);
+            var convertedNode = _testConverter.Convert2Blazor(htmlNode);
+            var actualString = convertedNode.WriteTo();
+
+            Assert.AreEqual(expectedString, actualString);
+        }
+
+        [Test]
+        public void Convert2Blazor_Sets_ReadOnly_Attribute_Properly()
+        {
+            var htmlString = @"<asp:TextBox ID=""TestId"" Text=""Example Text"" ReadOnly=""true"">";
+            var expectedString = @"<input id=""TestId"" value=""Example Text"" readonly="""">";
```

Comment:
Does the element created in `expectedString` function as expected? My understanding is that the `readonly` attribute does not take an assignment, for example: 
`<input id=""TestId"" value=""Example Text"" readonly>`

We may need to create a new `ViewLayerControlAttribute` to handle this case.

Reply:
I believe this format is also valid for boolean attributes, according to the html standard (https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes) the `readonly` attribute can be written as `readonly`, `readonly=readonly`, or `readonly=""`. If you want I can still make the change to handling boolean attributes the first way for stylistic reasons

Reply:
Awesome, sounds good. Should be fine as-is, thanks!